CONTRIBUTING.md

```diff
@@ -73,3 +82,22 @@ To verify that this file is up to date execute:
 ```bash
 go run ./cmd/licensecheck
 ```
+
+## Release process
+
+If you're a Temporal engineer / code owner, here's how to do a release:
+
+1. Do some manual testing to make sure things look good--start a dev-server, run
+   some workflows, etc.
```

Comment:
Shouldn't the automated tests cover this?
I would rather we replace this with "verify that CI passes for the commit hash".

Reply:
(Which I see that you have below).

Reply:
Yeah, the problem is nobody seems to fully-trust that CI is "good enough" right now—Chad's told me multiple times we need to do manual testing on things like server updates.

Reply:
It's fine as long as there's a step to verify that CI is green.